kong/clustering/rpc/socket.lua

```diff
@@ -151,7 +162,7 @@ function _M:start()
         ngx_log(ngx_DEBUG, "[rpc] got RPC call: ", payload.method, " (id: ", payload.id, ")")
 
         local dispatch_cb = self.manager.callbacks.callbacks[payload.method]
-        if not dispatch_cb then
+        if not dispatch_cb and payload.id then
```

Comment:
From the Json-RPC 2.0 spec, I think the reply MUST contain `id` field. So could we check whether there is a `payload.id` right after the line `  local payload = decompress_payload(data)`

Reply:
like 

```
local payload = decompress_payload(data)

-- check id firstly
if not payload.id then
   return error_log("not found id")
end

if type(payload.id) ~= number then
   return error_log("invalid id: not a number")
end

-- From now on, we assure that payload.id exists and it is a number
...
```

Reply:
No, the examles in json rpc spec:
```
--> {"jsonrpc": "2.0", "method": "update", "params": [1,2,3,4,5]}
--> {"jsonrpc": "2.0", "method": "foobar"}
```